### PR TITLE
Publish master Docker image to GPR

### PR DESCRIPTION
**Pull request by jMarkP**
Add a workflow to publish the docker image on push to master

### CODE REVIEW

File: .github/workflows/cd.yaml [hunk 1]
```diff
@@ -0,0 +1,88 @@
+name: cd
```
Comment (jMarkP) [inline .github/workflows/cd.yaml:1]: Most of this is Copy/Paste from the existing ci workflow

File: .github/workflows/cd.yaml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    - name: Download cache
+      uses: actions/download-artifact@v1
+      with:
+        name: cache
+    - name: Load cache
+      run: |        
+        docker image load --input cache/flux-latest.tar
+        docker image tag docker.pkg.github.com/$GITHUB_REPOSITORY/flux:$GITHUB_SHA
+        docker login docker.pkg.github.com -u $GITHUB_ACTOR -p ${{ secrets.GITHUB_TOKEN }}
+        docker push docker.pkg.github.com/$GITHUB_REPOSITORY/flux:$GITHUB_SHA
```
Comment (jMarkP) [inline .github/workflows/cd.yaml]: This is the new stuff:
- Wait for `e2e-testing` to pass
- Reload the docker image that got built in the `unit-testing` job
- Tag it
- Authenticate with docker.pkg.github
- Push

**Review by HineshMandalia — DISMISSED** [review]
Looks good to me

File: .github/workflows/cd.yaml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Download cache
+      uses: actions/download-artifact@v1
+      with:
+        name: cache
+    - name: Load cache
+      run: |        
+        docker image load --input cache/flux-latest.tar
+        docker image tag docker.pkg.github.com/$GITHUB_REPOSITORY/flux:$GITHUB_SHA
```
Comment (herebebeasties) [inline .github/workflows/cd.yaml]: Might be worth putting a date stamp in here, so that they are ordered? Or figure out the git "commit height" and use that as a version number?

File: .github/workflows/cd.yaml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  push:
+    branches:
+      - master
+
+jobs:
+  unit-testing:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: niden/actions-memcached@v7
```
Comment (herebebeasties) [inline .github/workflows/cd.yaml:13]: Nice to add a comment explaining why this is needed.

**Review by herebebeasties — DISMISSED** [review]

File: .github/workflows/cd.yaml (hunk 2 above)
Comment (herebebeasties) [reply .github/workflows/cd.yaml]: Bit hard to follow how this works with the cache stuff. Is this really the best way? Is this branch/thread-safe? Is latest cache always pulled from the master build here? What if you merge branches out of order that they are built?

File: .github/workflows/cd.yaml (hunk 3 above)
Comment (jMarkP) [reply .github/workflows/cd.yaml]: :+1: Will take a look at options

File: .github/workflows/cd.yaml (hunk 4 above)
Comment (jMarkP) [reply .github/workflows/cd.yaml:13]: This is copy-paste from flux's existing build workflow

File: .github/workflows/cd.yaml (hunk 2 above)
Comment (jMarkP) [reply .github/workflows/cd.yaml]: I think their use of 'cache' here is misleading - this seems to be like a Bamboo artifact that can be shared between workflow steps: https://help.github.com/en/actions/configuring-and-managing-workflows/persisting-workflow-data-using-artifacts
